### PR TITLE
Support for Kubernetes v1.25

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area open-source usability
/kind enhancement
/platform aws
/exp intermediate
/topology garden seed shoot
/merge squash

**What this PR does / why we need it**:
This PR adds support for Kubernetes 1.25 to the extension.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/6567

**Special notes for your reviewer**:
* ~Depends on #609 , hence, PR is in draft state.~
* ~Depends on #626~ 
* I have successfully validated the functionality as follows:
  * :white_check_mark: Create new clusters with versions < 1.25
  * :white_check_mark: Create new clusters with version  = 1.25
  * :white_check_mark: Upgrade old clusters from version 1.24 to version 1.25
  * :white_check_mark: Delete clusters with versions < 1.25
  * :white_check_mark: Delete clusters with version  = 1.25

**Release note**:

```feature user
The aws extension does now support shoot clusters with Kubernetes version 1.25. You should consider the [Kubernetes release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.25.md) before upgrading to 1.25. 
```

### CODE REVIEW

File: charts/gardener-extension-admission-aws/charts/runtime/templates/poddisruptionbudget.yaml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {{- if gt (int .Values.global.replicaCount) 1 }}
+{{- if semverCompare ">= 1.21-0" .Capabilities.KubeVersion.GitVersion }}
```
Comment (dkistner) [inline charts/gardener-extension-admission-aws/charts/runtime/templates/poddisruptionbudget.yaml:2]: Do we need also to allow prereleases?

**Comment by shafeeqes** [comment]
> Testrun: [e2e-rckpl](https://tm.gardener.cloud/testrun/default/e2e-rckpl) Workflow: e2e-rckpl-wf Phase: Failed
> 
> +---------------------+---------------------+--------+----------+
> |        NAME         |        STEP         | PHASE  | DURATION |
> +---------------------+---------------------+--------+----------+
> | infrastructure-test | infrastructure-test | Failed | 1m59s    |
> | bastion-test        | bastion-test        | Failed | 1m53s    |
> | dnsrecord-test      | dnsrecord-test      | Failed | 2m0s     |
> +---------------------+---------------------+--------+----------+

```
2022-10-07 17:54:41 | 2022-10-07T12:24:41.103706799Z stdout F FAIL	github.com/gardener/gardener-extension-provider-aws/test/integration/bastion [build failed]
-- | --
  |   | 2022-10-07 17:53:51 | 2022-10-07T12:23:51.55575366Z stderr F note: module requires Go 1.19
  |   | 2022-10-07 17:53:51 | 2022-10-07T12:23:51.555742345Z stderr F vendor/k8s.io/apimachinery/pkg/util/diff/diff.go:138:28: undefined: reflect.Pointer
  |   | 2022-10-07 17:53:51 | 2022-10-07T12:23:51.555698945Z stderr F # k8s.io/apimachinery/pkg/util/diff
  |   | 2022-10-07 17:53:47 | 2022-10-07T12:23:47.455806591Z stderr F note: module requires Go 1.19
  |   | 2022-10-07 17:53:47 | 2022-10-07T12:23:47.45579816Z stderr F vendor/k8s.io/apimachinery/third_party/forked/golang/json/fields.go:186:40: undefined: reflect.Pointer
  |   | 2022-10-07 17:53:47 | 2022-10-07T12:23:47.455789711Z stderr F vendor/k8s.io/apimachinery/third_party/forked/golang/json/fields.go:31:17: undefined: reflect.Pointer
  |   | 2022-10-07 17:53:47 | 2022-10-07T12:23:47.45574029Z stderr F # k8s.io/apimachinery/third_party/forked/golang/json
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.637104845Z stderr F note: module requires Go 1.19
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.637097438Z stderr F vendor/k8s.io/apimachinery/pkg/util/validation/field/errors.go:69:33: undefined: reflect.Pointer
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.637069101Z stderr F # k8s.io/apimachinery/pkg/util/validation/field
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535644354Z stderr F note: module requires Go 1.18
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535638704Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:175:22: too many errors
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535634447Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:175:22: undefined: any
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535630015Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:158:16: undefined: any
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535625221Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:1261:42: undefined: any
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535620852Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:1195:52: undefined: any
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535615552Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:1160:42: undefined: any
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535609736Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:1143:45: undefined: any
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535602507Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:937:48: undefined: any
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535597945Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:428:37: undefined: any
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535593144Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:177:35: undefined: any
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535587871Z stderr F vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:96:31: undefined: any
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.535553319Z stderr F # sigs.k8s.io/json/internal/golang/encoding/json
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.158866451Z stderr F note: module requires Go 1.19
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.158861272Z stderr F vendor/k8s.io/apimachinery/pkg/conversion/queryparams/convert.go:142:7: undefined: reflect.Pointer
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.158855209Z stderr F vendor/k8s.io/apimachinery/pkg/conversion/queryparams/convert.go:58:17: undefined: reflect.Pointer
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:11.158832596Z stderr F # k8s.io/apimachinery/pkg/conversion/queryparams
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:10.041024585Z stderr F note: module requires Go 1.19
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:10.041018222Z stderr F vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:376:7: undefined: reflect.Pointer
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:10.040994107Z stderr F # k8s.io/apimachinery/third_party/forked/golang/reflect
  |   | 2022-10-07 17:53:14 | 2022-10-07T12:23:08.164186427Z stderr F time="2022-10-07T12:23:08.163Z" level=info msg="capturing logs" argo=true
```
Probably we should get #602 in, and update the pipeline definitions?

**Review by dkistner — COMMENTED** [review]
looks good to me.
@kon-angelo is currently working on a pr to upgrade the golang version to v1.19 which will fix the integration tests. Once we have this we can move on and merge this one here.

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**